src/calls.c

```diff
@@ -1543,12 +1533,27 @@ Obj FuncCALL_FUNC_LIST (
         result = CALL_XARGS( func, list2 );
       }
     } else {
-      result = DoOperation2Args(self, func, list);
+      result = DoOperation2Args(CallFuncListOper, func, list);
     }
     /* return the result                                                   */
     return result;
+
 }
 
+Obj FuncCALL_FUNC_LIST (
+    Obj                 self,
+    Obj                 func,
+    Obj                 list )
+{
+  /* check that the second argument is a list                            */
+    while ( ! IS_SMALL_LIST( list ) ) {
+        list = ErrorReturnObj(
+            "CallFuncList: <list> must be a small list",
+            0L, 0L,
+            "you can replace <list> via 'return <list>;'" );
+    }
+    return CallFuncList(func, list);
+}
 
```

Comment:
You're right of course. I need to get out of the habit to doing tidying and minor refactoring as I "run into them" rather than as separate commits. For now I'll adjust the commit message.